zapcore/encoder.go

```diff
@@ -270,6 +270,18 @@ func (e *DurationEncoder) UnmarshalText(text []byte) error {
 	return nil
 }
 
+// A StacktraceEncoder serializes a stacktrace
+//
+// This function must make exactly one call
+// to a PrimitiveArrayEncoder's Append* method.
+type StacktraceEncoder func(string, PrimitiveArrayEncoder)
+
+// FullStacktraceEncoder passes down the full stacktrace as a string to the enc
+func FullStacktraceEncoder(stacktrace string, enc PrimitiveArrayEncoder) {
+	// TODO: consider using a byte-oriented API to save an allocation.
```

Comment:
FYI: This TODO will be unfixable once this change is released and these APIs are made public. We can discuss this more in the PR if you'd like, but I think this AppendString won't cause an alloc. (We've already paid for the stacktrace alloc elsewhere. )